type-definitions/almin.js.flow

```diff
@@ -0,0 +1,88 @@
+/**
+ * This file provides type definitions for use with the Flow type checker.
+ *
+ * @flow
+ */
+
+type DispatcherPayload = Object;
+type UseCaseErrorPayload = Object;
+
+declare class QueuedStoreGroup extends Dispatcher {
+    hasChangingStore(): boolean;
+    currentChangingStores(): Array<Store>;
+
+    getState(): Object;
+    emitChange(): void;
+    onChange(handler: (stores: Array<Store>) => mixed): Function;
+    release(): void;
+}
+
+declare class StoreGroup extends Dispatcher {
+    getState(): Object;
+    emitChange(): void;
+    onChange(handler: (stores: Array<Store>) => mixed): Function;
+    release(): void;
+}
+
+declare class Context {
+    getState(): mixed;
+    onChange(onChangeHandler: (changingStores: Array<Store>) => mixed): Function;
+    useCase(useCase: UseCase): UseCaseExecutor;
+    onWillExecuteEachUseCase(handler: (useCase: UseCase, args: mixed) => mixed): void;
+    onDispatch(handler: (payload: DispatcherPayload) => mixed): Function;
+    onDidExecuteEachUseCase(handler: (useCase: UseCase) => mixed): void;
+    onCompleteExecuteEachUseCase(handler: (useCase: UseCase) => mixed): void;
+    onErrorDispatch(errorHandler: (payload: UseCaseErrorPayload) => mixed): (this: Dispatcher) => mixed;
+    release(): void;
+}
+
+declare class Dispatcher extends EventEmitter {
+    static isDispatcher(maybeDispatcher: mixed): boolean;
+
+    onDispatch(payloadHandler: (payload: DispatcherPayload) => mixed): Function;
+    dispatch(payload: DispatcherPayload): void;
+    pipe(toDispatcher: Dispatcher): Function;
+}
+
+declare class Store extends Dispatcher {
+    static isStore(maybeStore: mixed): boolean;
+
+    getState(prevState: Object): Object;
+    onError(handler: (payload: UseCaseErrorPayload) => mixed): Function;
+    onChange(callback: Function): Function;
+    emitChange(): void;
+}
+
+declare class UseCase extends Dispatcher {
+    static isUseCase(maybeUseCase: mixed): boolean;
+
+    context(): UseCaseContext;
+    onError(errorHandler: (error: Error) => mixed): (this: Dispatcher) => mixed;
+    throwError(error: Error): void;
+}
+
+declare class UseCaseContext {
+    useCase(useCase: UseCase): UseCaseExecutor;
+}
+
+declare class UseCaseExecutor {
+    willExecute(args: Array<mixed>): void;
+    didExecute(): void;
+    complete(): void;
+    onWillExecuteEachUseCase(handler: (useCase: UseCase, args: mixed) => mixed): void;
+    onDidExecuteEachUseCase(handler: (useCase: UseCase) => mixed): void;
+    onCompleteExecuteEachUseCase(handler: (useCase: UseCase) => mixed): Function;
+    execute(args: Array<mixed>): void;
+    release(): void;
+}
+
+export {
+    Dispatcher,
+    Store,
+    StoreGroup,
+    QueuedStoreGroup,
+    UseCaseExecutor,
```

Comment:
📝 Currently,  `UseCaseExecutor` stay in the closet.
It means that `UseCaseExecutor` is under the unstable. It is potential to change.
But, I know that type definition(d.ts and flow) need to public the internal type because it can be touched from outside.


I forget  correctly JSDoc typing of `UseCaseExecutor`.
```diff
+    onWillExecuteEachUseCase(handler: (useCase: UseCase, args: mixed) => mixed): void;
+    onDidExecuteEachUseCase(handler: (useCase: UseCase) => mixed): void;
+    onCompleteExecuteEachUseCase(handler: (useCase: UseCase) => mixed): Function;
+    execute(args: Array<mixed>): void;
+    release(): void;
```

are public.

```diff
+    willExecute(args: Array<mixed>): void;
+    didExecute(): void;
+    complete(): void;
```

are private.